### PR TITLE
pythonPackages.labelbox: init at 2.1

### PR DESCRIPTION
**Pull request by Rakesh4G**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @CMCDragonkai

### CODE REVIEW

**Comment by mmahut** [comment]
@GrahamcOfBorg build pythonPackages.labelbox python3Packages.labelbox

File: pkgs/development/python-modules/labelbox/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/labelbox/default.nix]:
```suggestion
  propagatedBuildInputs = [ requests ];
```

File: pkgs/development/python-modules/labelbox/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/labelbox/default.nix]: i would remove all the inputs you dont need, in this case you should just need `lib, buildPythonPackage, fetchPypi, requests`

File: pkgs/development/python-modules/labelbox/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/labelbox/default.nix]: for buildPythonPackage, doCheck is defaulted to true, you can remove this
```suggestion
```

**Comment by Rakesh4G** [comment]
@jonringer , Thanks i will test the suggested changes and update accordingly.

File: pkgs/development/python-modules/labelbox/default.nix
Comment (Rakesh4G) [reply pkgs/development/python-modules/labelbox/default.nix]: Done.

File: pkgs/development/python-modules/labelbox/default.nix
Comment (Rakesh4G) [reply pkgs/development/python-modules/labelbox/default.nix]: Done.

File: pkgs/development/python-modules/labelbox/default.nix
Comment (Rakesh4G) [reply pkgs/development/python-modules/labelbox/default.nix]: Done.

**Comment by Rakesh4G** [comment]
@jonringer , I have updated the suggested changes and tested.

@GrahamcOfBorg build pythonPackages.labelbox python3Packages.labelbox

**Comment by jonringer** [comment]
you need to be whitelisted for ofborg to pick up commands

@GrahamcOfBorg build pythonPackages.labelbox python3Packages.labelbox

**Comment by Rakesh4G** [comment]
> you need to be whitelisted for ofborg to pick up commands
> 
> @GrahamcOfBorg build pythonPackages.labelbox python3Packages.labelbox

OK @jonringer , Thanks.

**Comment by jonringer** [comment]
I noticed that there were no tests ran, it appears that the pypi package doesn't include tests. I would pull source from https://github.com/Labelbox/labelbox-python so that tests are ran (this will help to know if the package is broken by one of it's dependencies being updated)

**Comment by Rakesh4G** [comment]
> I noticed that there were no tests ran, it appears that the pypi package doesn't include tests. I would pull source from https://github.com/Labelbox/labelbox-python so that tests are ran (this will help to know if the package is broken by one of it's dependencies being updated)

OK @jonringer . I will update this.

**Comment by Rakesh4G** [comment]
> I noticed that there were no tests ran, it appears that the pypi package doesn't include tests. I would pull source from https://github.com/Labelbox/labelbox-python so that tests are ran (this will help to know if the package is broken by one of it's dependencies being updated)

Hi @jonringer , I had tried to take source from https://github.com/Labelbox/labelbox-python 

- release source on GitHub is showing version 0.0.2 (9 Sep 2018) . On https://pypi.org labelbox 2.0.1 (3 Sep 2019) is available. Also 
- when i tried to build labelbox-python from GitHub it used pipenv and is causing access related issues. not sure how to proceed on this.

 
`Traceback (most recent call last):
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/bin/.pipenv-wrapped", line 11, in <module>
    sys.exit(cli())
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/cli/command.py", line 408, in run
    command=command, args=args, three=state.three, python=state.python, pypi_mirror=state.pypi_mirror
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/core.py", line 2310, in do_run
    three=three, python=python, validate=False, pypi_mirror=pypi_mirror,
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/core.py", line 574, in ensure_project
    pypi_mirror=pypi_mirror,
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/core.py", line 489, in ensure_virtualenv
    if not project.virtualenv_exists:
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/project.py", line 272, in virtualenv_exists
    if self.pipfile_exists and os.path.exists(self.virtualenv_location):
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/project.py", line 437, in virtualenv_location
    self._virtualenv_location = self.get_location_for_virtualenv()
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/project.py", line 294, in get_location_for_virtualenv
    return str(get_workon_home().joinpath(self.virtualenv_name))
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py", line 1414, in get_workon_home
    mkdir_p(str(expanded_path))
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py", line 864, in mkdir_p
    mkdir_p(head)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py", line 864, in mkdir_p
    mkdir_p(head)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py", line 864, in mkdir_p
    mkdir_p(head)
  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py", line 869, in mkdir_p
    os.mkdir(newdir)
PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
`

**Comment by CMCDragonkai** [comment]
I think you can use the master branch with a fixed commit.

On 11 September 2019 19:27:53 GMT+10:00, Rakesh Gupta <notifications@github.com> wrote:
>> I noticed that there were no tests ran, it appears that the pypi
>package doesn't include tests. I would pull source from
>https://github.com/Labelbox/labelbox-python so that tests are ran (this
>will help to know if the package is broken by one of it's dependencies
>being updated)
>
>Hi @jonringer , I had tried to take source from
>https://github.com/Labelbox/labelbox-python 
>
>- release source on GitHub is showing version 0.0.2 (9 Sep 2018) . On
>https://pypi.org labelbox 2.0.1 (3 Sep 2019) is available. Also 
>- when i tried to build labelbox-python from GitHub it used pipenv and
>is causing access related issues. not sure how to proceed on this.
>
> 
>`Traceback (most recent call last):
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/bin/.pipenv-wrapped",
>line 11, in <module>
>    sys.exit(cli())
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py",
>line 764, in __call__
>    return self.main(*args, **kwargs)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py",
>line 717, in main
>    rv = self.invoke(ctx)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py",
>line 1137, in invoke
>    return _process_result(sub_ctx.command.invoke(sub_ctx))
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py",
>line 956, in invoke
>    return ctx.invoke(self.callback, **ctx.params)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py",
>line 555, in invoke
>    return callback(*args, **kwargs)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/decorators.py",
>line 64, in new_func
>    return ctx.invoke(f, obj, *args, **kwargs)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/vendor/click/core.py",
>line 555, in invoke
>    return callback(*args, **kwargs)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/cli/command.py",
>line 408, in run
>command=command, args=args, three=state.three, python=state.python,
>pypi_mirror=state.pypi_mirror
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/core.py",
>line 2310, in do_run
>   three=three, python=python, validate=False, pypi_mirror=pypi_mirror,
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/core.py",
>line 574, in ensure_project
>    pypi_mirror=pypi_mirror,
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/core.py",
>line 489, in ensure_virtualenv
>    if not project.virtualenv_exists:
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/project.py",
>line 272, in virtualenv_exists
>   if self.pipfile_exists and os.path.exists(self.virtualenv_location):
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/project.py",
>line 437, in virtualenv_location
>    self._virtualenv_location = self.get_location_for_virtualenv()
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/project.py",
>line 294, in get_location_for_virtualenv
>    return str(get_workon_home().joinpath(self.virtualenv_name))
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py",
>line 1414, in get_workon_home
>    mkdir_p(str(expanded_path))
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py",
>line 864, in mkdir_p
>    mkdir_p(head)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py",
>line 864, in mkdir_p
>    mkdir_p(head)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py",
>line 864, in mkdir_p
>    mkdir_p(head)
>File
>"/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py",
>line 869, in mkdir_p
>    os.mkdir(newdir)
>PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
>`
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/68097#issuecomment-530299851

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jonringer** [comment]
> release source on GitHub is showing version 0.0.2 (9 Sep 2018) . On https://pypi.org labelbox 2.0.1 (3 Sep 2019) is available. 

interesting... I wonder how they are doing releases then...

> when i tried to build labelbox-python from GitHub it used pipenv and is causing access related issues. not sure how to proceed on this

So, $HOME is usually set to "/homeless-shelter", and this is intentional so that the builds aren't leaking into the system. generally I do `HOME=$TMPDIR` for things like tests, but for builds im not sure what the right answer is, as it may need some of those artifacts later. I also don't have much experience with pipenv builds.

**Comment by Rakesh4G** [comment]
> I think you can use the master branch with a fixed commit.

Permission related error is still there on my system. 

`  File "/nix/store/m7kvd3qg2dwl6kzp583f6bdc16d2hpps-pipenv-2018.11.26/lib/python3.7/site-packages/pipenv/utils.py", line 869, in mkdir_p
    os.mkdir(newdir)
PermissionError: [Errno 13] Permission denied: '/homeless-shelter'
builder for '/nix/store/hcgf2bi6pyaj3giz9isxwjnin003agbf-labelbox-python-0.0.2.drv' failed with exit code 1
`

File: pkgs/development/python-modules/labelbox/default.nix
Comment (Rakesh4G) [reply pkgs/development/python-modules/labelbox/default.nix]: Updated as per suggestion.

File: pkgs/development/python-modules/labelbox/default.nix
Comment (CMCDragonkai) [inline pkgs/development/python-modules/labelbox/default.nix]: Put in a comment as to why you are disabling the tests.

Comment (CMCDragonkai) [reply pkgs/development/python-modules/labelbox/default.nix]: Since the tests aren't working from the Github source anyway, maybe we can revert back to PyPi? @jonringer.

**Comment by CMCDragonkai** [comment]
If everything works, at the end make sure to squash all commits into 1 commit following the commit guidelines for nixpkgs.

**Comment by Rakesh4G** [comment]
> If everything works, at the end make sure to squash all commits into 1 commit following the commit guidelines for nixpkgs.

Hi @CMCDragonkai , I have squashed all commits into 1 commit. I think this is now ready for merge from my side. Thanks.

File: pkgs/development/python-modules/labelbox/default.nix
Comment (Rakesh4G) [reply pkgs/development/python-modules/labelbox/default.nix]: I have reverted this back to PyPi.

**Comment by CMCDragonkai** [comment]
There's another MR with adding Rakesh into contributor list. Should we wait for that be merged first?

On 31 October 2019 02:34:46 GMT+11:00, Jon <notifications@github.com> wrote:
>jonringer commented on this pull request.
>
>
>
>> +  rakesh4g = {
>+    email = "rakeshgupta4u@gmail.com";
>+    github = "Rakesh4G";
>+    name = "Rakesh Gupta";
>+  };
>
>please add githubId and put this into a different commit
>
>This will also cause a merge conflict with your other PR
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/NixOS/nixpkgs/pull/68097#pullrequestreview-309301772

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jonringer** [comment]
which one is able to be merged, I'll merge fist

**Comment by Rakesh4G** [comment]
> which one is able to be merged, I'll merge fist
Hi @jonringer ,
https://github.com/NixOS/nixpkgs/pull/67760 is good to go first.

**Comment by jonringer** [comment]
please remove maintainers commit :)

**Comment by Rakesh4G** [comment]
> please remove maintainers commit :)

OK, Thanks.

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.labelbox python37Packages.labelbox python38Packages.labelbox

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
commits LGTM
outputs LGTM

```
[5 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/68097
3 package were build:
python27Packages.labelbox python37Packages.labelbox python38Packages.labelbox
```

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
commits LGTM

```
[5 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/68097
3 package were build:
python27Packages.labelbox python37Packages.labelbox python38Packages.labelbox
```

**Pull request merged**
